### PR TITLE
FEAT: Suppress from Nav

### PR DESCRIPTION
**Pull request by Flowdeeps**
Added a checkbox which will allow user to not have a section automatically appear in the site nav. I know it bugs the heck out of me.

### CODE REVIEW

File: models/storyboard.js [hunk 1]
```diff
@@ -19,14 +19,22 @@ function createNav(items) {
     const nav = [];
 
     items.forEach((item, id) => {
-        const [mediaType] = Object.keys(item);
-        // fallback for now until every type has a title field to fill in. (or something for nav)
-        const tmpTitle = item[mediaType].title ? item[mediaType].title : `${id} ${mediaType}`;
-        const title = tmpTitle.replace(/<\/?[^>]+(>|$)/g, "");
+      var elKey = Object.keys(item)[0];
```
Comment (naomiaro) [inline models/storyboard.js]: elKey is the same as mediaType below I think

File: models/storyboard.js [hunk 2]
```diff
@@ -19,14 +19,22 @@ function createNav(items) {
...
+      const root = item[elKey];
+      var writeToNav = true;
+      if (root && root.in_nav && root.in_nav.suppress) {
```
Comment (naomiaro) [inline models/storyboard.js]: can we not have this so nested? Why not just store a boolean right on in_nav?

File: models/storyboard.js [hunk 3]
```diff
@@ -19,14 +19,22 @@ function createNav(items) {
...
+        writeToNav = false;
+      }
+      const [mediaType] = Object.keys(item);
+      // fallback for now until every type has a title field to fill in. (or something for nav)
+      const tmpTitle = item[mediaType].title ? item[mediaType].title : `${id} ${mediaType}`;
+      const title = tmpTitle.replace(/<\/?[^>]+(>|$)/g, "");
```
Comment (naomiaro) [inline models/storyboard.js]: this might leave garbage in the title, if it's html better use innerText

Comment (Flowdeeps) [reply models/storyboard.js]: I've tried this but it's actually just a text string objects with html like entities in. This also strips all MS:O guff as well. I'd be happy for someone to take it over and sanitise if they wanted.

File: models/storyboard.js (hunk 1 above)
Comment (Flowdeeps) [reply models/storyboard.js]: Noted and acted upon

File: models/storyboard.js [hunk 4]
```diff
@@ -21,14 +21,20 @@ function createNav(items) {
     const nav = [];
 
     items.forEach((item, id) => {
-        const [mediaType] = Object.keys(item);
-        // fallback for now until every type has a title field to fill in. (or something for nav)
-        const tmpTitle = item[mediaType].title ? item[mediaType].title : `${id} ${mediaType}`;
-        const title = tmpTitle.replace(/<\/?[^>]+(>|$)/g, "");
+      const [mediaType] = Object.keys(item);
+      var writeToNav = true;
+      if (item[mediaType] && item[mediaType].in_nav) {
```
Comment (naomiaro) [inline models/storyboard.js]: shouldn't `in_nav` here be `true`? Why do you set `writeToNav`` false`?

File: routes/editor.js [hunk 5]
```diff
@@ -422,6 +430,9 @@ router.post('/page/imagebackground/id/:id', function (req, res) {
         // format and save new item
         var fullpage = (newItem['fullpage'] === 'on') ? true : false;
         item['format'] = { fullpage: fullpage };
+        var suppress = (newItem['suppress'] === 'on') ? true : false;
+        item['in_nav'] = suppress;
```
Comment (naomiaro) [inline routes/editor.js]: let's make in_nav true if it should be in the nav...

File: models/storyboard.js (hunk 2 above)
Comment (Flowdeeps) [reply models/storyboard.js]: Done

File: routes/editor.js (hunk 5 above)
Comment (Flowdeeps) [reply routes/editor.js]: It's on by default so it's more the fact that it should be reversed. I've changed `in_nav` to just `suppress` which although means the word loses all meaning after the amount of times I've seen it but at least it has an inner logic which is more obvious to an outsider.

File: models/storyboard.js (hunk 4 above)
Comment (Flowdeeps) [reply models/storyboard.js]: Explained beneath.

**Pull request merged**